### PR TITLE
feature : support zip bzip2 7z compressor

### PR DESCRIPTION
**Pull request by ph3636**

### CODE REVIEW

File: compressor/seata-compressor-7z/pom.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>seata-core</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.tukaani</groupId>
+            <artifactId>xz</artifactId>
+            <version>1.8</version>
```
Comment (xingfudeshi) [inline compressor/seata-compressor-7z/pom.xml]: Please maintain dependency version in bom/pom.xml

File: compressor/seata-compressor-7z/pom.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-compress</artifactId>
+            <version>1.19</version>
```
Comment (xingfudeshi) [inline compressor/seata-compressor-7z/pom.xml]: Please maintain dependency version in bom/pom.xml

File: compressor/seata-compressor-zip/src/main/java/io/seata/compressor/zip/ZipUtil.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        }
+    }
+
+    public static byte[] decompress(byte[] bytes) {
+        if (bytes == null) {
+            throw new NullPointerException("bytes is null");
+        }
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            ZipInputStream zip = new ZipInputStream(new ByteArrayInputStream(bytes));
```
Comment (xingfudeshi) [inline compressor/seata-compressor-zip/src/main/java/io/seata/compressor/zip/ZipUtil.java]: Pls use try with resource statement to close ZipInputStream .

File: compressor/seata-compressor-zip/src/main/java/io/seata/compressor/zip/ZipUtil.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    private static final int BUFFER_SIZE = 8192;
+
+    public static byte[] compress(byte[] bytes) {
+        if (bytes == null) {
+            throw new NullPointerException("bytes is null");
+        }
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            ZipOutputStream zip = new ZipOutputStream(out);
```
Comment (xingfudeshi) [inline compressor/seata-compressor-zip/src/main/java/io/seata/compressor/zip/ZipUtil.java]: Pls use try with resource statement to close ZipOutputStream .

File: compressor/seata-compressor-bzip2/src/main/java/io/seata/compressor/bzip2/BZip2Util.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    public static byte[] decompress(byte[] bytes) {
+        if (bytes == null) {
+            throw new NullPointerException("bytes is null");
+        }
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            ByteArrayInputStream bis = new ByteArrayInputStream(bytes);
+            CBZip2InputStream bzip2 = new CBZip2InputStream(bis);
```
Comment (xingfudeshi) [inline compressor/seata-compressor-bzip2/src/main/java/io/seata/compressor/bzip2/BZip2Util.java]: Pls use try with resource statement to close CBZip2InputStream .

File: compressor/seata-compressor-bzip2/src/main/java/io/seata/compressor/bzip2/BZip2Util.java [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+    private static final int BUFFER_SIZE = 8192;
+
+    public static byte[] compress(byte[] bytes) {
+        if (bytes == null) {
+            throw new NullPointerException("bytes is null");
+        }
+        try {
+            ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            CBZip2OutputStream bzip2 = new CBZip2OutputStream(bos);
```
Comment (xingfudeshi) [inline compressor/seata-compressor-bzip2/src/main/java/io/seata/compressor/bzip2/BZip2Util.java]: Pls use try with resource statement to close CBZip2OutputStream .

File: compressor/seata-compressor-bzip2/pom.xml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    <dependencies>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>seata-core</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.ant</groupId>
+            <artifactId>ant</artifactId>
+            <version>1.10.6</version>
```
Comment (xingfudeshi) [inline compressor/seata-compressor-bzip2/pom.xml]: Please maintain dependency version in bom/pom.xml

File: compressor/seata-compressor-7z/src/main/java/io/seata/compressor/sevenz/SevenZUtil.java [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    public static byte[] decompress(byte[] bytes) {
+        if (bytes == null) {
+            throw new NullPointerException("bytes is null");
+        }
+        try {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            SeekableInMemoryByteChannel channel = new SeekableInMemoryByteChannel(bytes);
+            SevenZFile sevenZFile = new SevenZFile(channel);
```
Comment (xingfudeshi) [inline compressor/seata-compressor-7z/src/main/java/io/seata/compressor/sevenz/SevenZUtil.java]: Pls use try with resource statement to close SevenZFile .

File: compressor/seata-compressor-7z/src/main/java/io/seata/compressor/sevenz/SevenZUtil.java [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+    private static final int BUFFER_SIZE = 8192;
+
+    public static byte[] compress(byte[] bytes) {
+        if (bytes == null) {
+            throw new NullPointerException("bytes is null");
+        }
+        try {
+            SeekableInMemoryByteChannel channel = new SeekableInMemoryByteChannel();
+            SevenZOutputFile z7z = new SevenZOutputFile(channel);
```
Comment (xingfudeshi) [inline compressor/seata-compressor-7z/src/main/java/io/seata/compressor/sevenz/SevenZUtil.java]: Pls use try with resource statement to close SevenZOutputFile .

**Comment by ph3636** [comment]
> Thanks.I left some comments.

done

**Comment by zjinlei** [comment]
test BZIP2 ZIP SEVENZ both failed
io.netty.handler.codec.CorruptedFrameException: Adjusted frame length (0) is less than lengthFieldEndOffset: 7

**Comment by ph3636** [comment]
> test BZIP2 ZIP SEVENZ both failed
> io.netty.handler.codec.CorruptedFrameException: Adjusted frame length (0) is less than lengthFieldEndOffset: 7
Unit test can pass. How to generate this? I don't quite understand

**Comment by ph3636** [comment]
> test BZIP2 ZIP SEVENZ both failed
> io.netty.handler.codec.CorruptedFrameException: Adjusted frame length (0) is less than lengthFieldEndOffset: 7

Unit test can pass. How to generate this? I don't quite understand

**Comment by zjinlei** [comment]
> > test BZIP2 ZIP SEVENZ both failed
> > io.netty.handler.codec.CorruptedFrameException: Adjusted frame length (0) is less than lengthFieldEndOffset: 7
> 
> Unit test can pass. How to generate this? I don't quite understand

need to configure them in seata-all pom.xml

**Pull request merged**